File: hw/codena/u3/hw21Interfaces/Python/AnimalFarmSystem/model/pig.py
from model.farm_animal import FarmAnimal
from model.cut import Cut

from controller.i_meat_animal import IMeatAnimal

class Pig(FarmAnimal, IMeatAnimal):

    def __init__(self, id, breed, born_on_date, weight, slaughter_house, product, cuts, ideal_weight):
        super.__init__(id, breed, born_on_date, weight, slaughter_house, product, cuts)
        self.ideal_weight = ideal_weight

    def feed(self, food):
        print(f"Feeding the PIG with food --> {food} in a bowl and water")

    def cut(self):

        cuts = []

        if self.slaughter_house.name is None:
            print("First send the Pig to slaughterHouse")

        else:

            cuts.append(Cut(1, "Ham", "Pig", 20))
            cuts.append(Cut(2, "Loin", "Pig", 16))
            cuts.append(Cut(3, "Belly", "Pig", 16.5))
            cuts.append(Cut(4, "Feet", "Pig", 5))

            print("Pig has been cut")

            for cut in cuts:
                print(cut.description)

        return cuts
    
    def send_to_slaughter_house(self, slaughter_house):
        if(self.weight >= self.ideal_weight):
            self.slaughter_house = slaughter_house
            print(f"Pig has been sent to SlaughterHouse {slaughter_house.name}")
        else:
            print("Pig has not reached the ideal weight, it was not sent to slaughterhouse")

File: hw/codena/u3/hw21Interfaces/JavaScript/AnimalFarmSystem/model/Pig.js
import FarmAnimal from "./FarmAnimal.js";
import Cut from "./Cut.js"

export default class Pig extends FarmAnimal {
    constructor(id, breed, bornOnDate, weight, slaughterHouse, product, cuts, idealWeight) {
        super(id, breed, bornOnDate, weight, slaughterHouse, product, cuts)
        this.idealWeight = idealWeight
    }

    feed(food) {
        console.log(`Feeding the Pig with ${food} in a bowl and water`)
    }

    cut() {
        let cuts = [];

        if (this.slaughterHouse == null || this.slaughterHouse.name == null) {
            console.log("First send the Pig to slaughterHouse");

        } else {

            cuts.push(new Cut(1, "Ham", "Pig", 20));
            cuts.push(new Cut(2, "Loin", "Pig", 16));
            cuts.push(new Cut(3, "Belly", "Pig", 16.5));
            cuts.push(new Cut(4, "Feet", "Pig", 5));

            console.log("Pig has been cut");

            for (const cut of cuts) {
                console.log(cut.description);
            }
        }

        return cuts;
    }

    sendToSlaughterHouse(slaughterHouse) {

        if (this.weight >= this.idealWeight) {
            this.slaughterHouse = slaughterHouse;
            console.log(`Pig has been sent to SlaughterHouse  ${slaughterHouse.name}`);

        } else {
            console.log("Pig has not reached the ideal weight, it was not sent to slaughterhouse");
        }
    }

    milk() {
        let milkProduced = this.weight / 100;
        return milkProduced;
    }

}
